src/WebJobs.Script/Diagnostics/ILoggerFactoryBuilder.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Diagnostics
+{
+    /// <summary>
+    /// Provides ways to plug into the ScriptHost ILoggerFactory initialization.
+    /// </summary>
+    public interface ILoggerFactoryBuilder
```

Comment:
I agree -- I'm open to suggestions.

It's not *quite* a Factory as the the ILoggerFactory is already created. This allows you to plug in additional LoggerProviders into our pipeline. We have our important ones (App Insights, FileLogger) that we don't want you to touch yet, but we do want to allow the CLI to plug in a ConsoleLogger. You can't just give us an ILoggerFactory that you've created yourself b/c then you don't have access to the Filtering settings that come from host.json. So I need to have a way to handle this after the ScriptConfig is built.